File: app/src/main/java/com/tobiasfried/rasa/messaging/MessageService.java
package com.tobiasfried.rasa.messaging;

import android.util.Log;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.messaging.FirebaseMessagingService;

public class MessageService extends FirebaseMessagingService {

    public static final String LOG_TAG = MessageService.class.getSimpleName();

    public static void getInstanceId() {
        FirebaseInstanceId.getInstance().getInstanceId()
                .addOnCompleteListener(task -> {
                    if (!task.isSuccessful()) {
                        Log.w(LOG_TAG, "getInstanceId failed", task.getException());
                        return;
                    }

                    // Get new Instance ID token
                    String token = task.getResult().getToken();

                    // Log and toast
                    Log.d(LOG_TAG, token);
                });
    }

    @Override
    public void onNewToken(String token) {
        Log.d(LOG_TAG, "Refreshed token: " + token);
    }
}

File: app/src/main/java/com/tobiasfried/rasa/messaging/MessageService.kt
package com.tobiasfried.rasa.messaging

import android.util.Log

import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.messaging.FirebaseMessagingService

class MessageService : FirebaseMessagingService() {

    override fun onNewToken(token: String) {
        Log.d(LOG_TAG, "Refreshed token: $token")
    }

    companion object {

        val LOG_TAG = MessageService::class.java.simpleName

        fun getInstanceId() {
            FirebaseInstanceId.getInstance().instanceId
                    .addOnCompleteListener { task ->
                        if (!task.isSuccessful) {
                            Log.w(LOG_TAG, "getInstanceId failed", task.exception)
                            return@FirebaseInstanceId.getInstance().getInstanceId()
                                    .addOnCompleteListener
                        }

                        // Get new Instance ID token
                        val token = task.result!!.token

                        // Log and toast
                        Log.d(LOG_TAG, token)
                    }
        }
    }
}
